### PR TITLE
[IMP] im_livechat,*: introduce LivechatButtonView/chatbotCurrentStep

### PR DESCRIPTION
**Pull request by alexkuhn**
*: website_livechat

Task-2888526

### CODE REVIEW

**Review by oomsveta — COMMENTED** [review]
need rebase

**Comment by alexkuhn** [comment]
> need rebase

Rebased

File: addons/im_livechat/static/src/legacy/public_livechat_chatbot.js [hunk 1]
```diff
@@ -434,20 +437,22 @@ const QWeb = core.qweb;
...
+                && !this.messaging.livechatButtonView.chatbotCurrentStep.is_email_valid
+            ) {
                 // email is not (yet) valid, let the user answer / try again
                 return false;
             } else if (
                 (this._chatbotIsExpectingUserInput() ||
-                this._chatbotCurrentStep.chatbot_step_type === 'question_selection') &&
-                this.messaging.livechatButtonView.messages.length !== 0
+                this.messaging.livechatButtonView.chatbotCurrentStep.chatbot_step_type === 'question_selection') &&
+                this._messages.length !== 0
```
Comment (oomsveta) [inline addons/im_livechat/static/src/legacy/public_livechat_chatbot.js:452]: why going back to `this._messages`?

File: addons/im_livechat/static/src/legacy/public_livechat_chatbot.js [hunk 2]
```diff
@@ -434,20 +437,22 @@ const QWeb = core.qweb;
...
             ) {
                 const lastMessage = this.messaging.livechatButtonView.messages[this.messaging.livechatButtonView.messages.length - 1];
-                if (lastMessage.getAuthorID() !== this.messaging.livechatButtonView.livechat.getOperatorPID()[0]) {
+                if (lastMessage.getAuthorID() !==  this.messaging.livechatButtonView.livechat.getOperatorPID()[0]) {
```
Comment (oomsveta) [inline addons/im_livechat/static/src/legacy/public_livechat_chatbot.js:455]: unwanted space

Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/public_livechat_chatbot.js]: changes done

**Comment by alexkuhn** [comment]
Merged in grouped PR: https://github.com/odoo/odoo/pull/94190

**Pull request closed without merging**